plugin.video.nba/nbatvlive.py

```diff
@@ -62,70 +107,129 @@ def playLiveTV():
-
-                if int(video_height) == vars.target_video_height:
-                    selected_video_path = streamdata.attributes["url"].value
-                    http_servers = streamdata.getElementsByTagName("httpserver")
-
-                    for http_server in http_servers:
-                        server_name = http_server.attributes["name"].value
-                        server_port = http_server.attributes["port"].value
-
-                        # Construct the video url directly in m3u8
-                        m3u8_video_url = "http://%s:%s%s.m3u8" % (server_name, server_port, selected_video_path)
-                        
-                        # Test if the video is actually available. If it is not available go to the next server.
-                        if urllib.urlopen(m3u8_video_url).getcode() == 200:
-                            video_url = m3u8_video_url
-
-                            # Get the cookies from the xml tag <encryption>
-                            video_cookies = streamdata.getElementsByTagName("encryption")[0].attributes['token'].value
-                            video_cookies = video_cookies.replace(';', '; ')
-                            video_cookies_encoded = urllib.quote(video_cookies)
-                            log("nba live tv: live cookie: %s" % video_cookies, xbmc.LOGDEBUG)
-                            break
-
-                        log("no working url found for this server, moving to the next", xbmc.LOGDEBUG)
-
-                    # break from the video quality loop
-                    break
-
-        # Add the cookies in the format "videourl|Cookie=[cookies]""
-        video_url = "%s?%s|Cookie=%s" % (video_url, querystring, video_cookies_encoded)
-
-    item = xbmcgui.ListItem("NBA TV Live", path=video_url)
-    xbmcplugin.setResolvedUrl(int(sys.argv[1]), True, item)
+            # Transform the link from adaptive://domain/url?querystring to
+            # http://domain/play?url=url&querystring
+            match = re.search('adaptive://([^/]+)(/[^?]+)\?(.+)$', video_adaptive_url)
+            domain = match.group(1)
+            path = urllib.quote_plus(str(match.group(2)))
+            querystring = match.group(3)
+            video_play_url = "http://%s/play?url=%s&%s" % (domain, path, querystring)
+            log("nba live tv: play url is %s" % video_play_url, xbmc.LOGDEBUG)
+
+            # Get the video play url (which will return different urls for
+            # different bitrates)
+            try:
+                request = urllib2.Request(video_play_url, None, {'Cookie': vars.cookies})
+                response = urllib2.urlopen(request)
+                content = response.read()
+            except urllib2.HTTPError as e:
+                log("nba live tv: failed getting url: %s %s" % (video_play_url, e.read()))
+                littleErrorPopup( xbmcaddon.Addon().getLocalizedString(50023) )
+                return
+
+            if not content:
```

Comment:
I'm checking whether a variable is an empty string or not, not whether it exists or not. In this case the http request can return an empty string, but the variable should be defined